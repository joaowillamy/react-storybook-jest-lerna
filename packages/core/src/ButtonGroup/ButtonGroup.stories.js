import React from "react";
import { ThemeProvider } from "@joaowillamy-test-quero/theme";

import ButtonGroup from ".";

export default {
  component: ButtonGroup,
  title: "Core|ButtonGroup"
};

export const buttonGroup = () => (
  <ThemeProvider>
    <ButtonGroup
      options={[
        { label: "Todos os semestres", value: "all" },
        { label: "2ยบ semestre de 2019", value: "second" },
        { label: "1ยบ semestre de 2020", value: "first" }
      ]}
      onChange={option => console.log("option", option)}
    ></ButtonGroup>
  </ThemeProvider>
);
