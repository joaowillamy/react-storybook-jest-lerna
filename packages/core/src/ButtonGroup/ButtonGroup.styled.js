import styled, { css } from "styled-components";
import media from "styled-media-query";

import { Button } from "../";

export const ButtonGroup = styled.div``;

export const ButtonGroupOption = styled(Button)`
  margin: 0;
  border-radius: 0;

  ${({ isSelected, theme }) =>
    isSelected &&
    css`
      background-color: ${theme.colors.blueSecondary};
      color: #fff;
    `};

  :first-child {
    border-radius: 5px 5px 0 0;

    ${media.greaterThan("medium")`
      border-radius: 5px 0 0 5px;
    `}
  }

  :last-child {
    border-radius: 0 0 5px 5px;

    ${media.greaterThan("medium")`
      border-radius: 0 5px 5px 0;
    `}
  }
`;
