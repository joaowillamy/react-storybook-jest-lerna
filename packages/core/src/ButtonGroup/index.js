import React, { useCallback, useState } from "react";
import PropTypes from "prop-types";

import * as S from "./ButtonGroup.styled";

const ButtonGroup = ({ onChange, options, ...props }) => {
  const [selected, setSelected] = useState("all");
  const callbackOnChange = useCallback(onChange);

  const changeSelectedOption = option => {
    setSelected(option);
    callbackOnChange && callbackOnChange(option);
  };

  return (
    <S.ButtonGroup {...props}>
      {options.map((option, index) => (
        <S.ButtonGroupOption
          key={index}
          onClick={() => changeSelectedOption(option.value)}
          isSelected={selected == option.value}
          variant="secondary"
        >
          {option.label}
        </S.ButtonGroupOption>
      ))}
    </S.ButtonGroup>
  );
};

ButtonGroup.propTypes = {
  options: PropTypes.array,
  onChange: PropTypes.func
};

ButtonGroup.defaultProps = {
  options: []
};

export default ButtonGroup;
