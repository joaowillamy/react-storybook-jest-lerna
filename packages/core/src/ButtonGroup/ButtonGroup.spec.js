import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";

import ButtonGroup from ".";

describe("ButtonGroup", () => {
  test("primary", () => {
    const tree = renderer.create(<ButtonGroup>{"Test"}</ButtonGroup>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
