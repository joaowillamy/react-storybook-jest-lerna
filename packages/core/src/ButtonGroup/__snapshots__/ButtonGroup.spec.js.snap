// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGroup primary 1`] = `
<div
  className=""
/>
`;
